### PR TITLE
feat: check pull requests and commits using conventional commits

### PR DESCRIPTION
**Pull request by ferdodo**
I added two new more workflows, the purpose is to have better consistency over the commit history and pull requests.

- The `./git-conventional-commits.yaml` configuration file is enforced on the latest commit of the pull request.
- Pull requests must respect a basic name and description regex based on conventional commit specification.
- Improved job and steps names for other workflows.

image

### CODE REVIEW

**Comment by Jean-BaptisteC** [comment]
Maybe use latest version of checkout and setup-node actions?
checkout v3 use a deprecated version of node js.

**Comment by ferdodo** [comment]
it is indeed using node16, I though it was supposed to take the node version from package.json as mentionned here https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#node-version-file , but you actually have to reference where the package.json is at.

I'll add that too.

**Comment by pbrissaud** [comment]
To avoid using a deprecated version, we can use a tool like [Renovate](https://www.mend.io/renovate/) to automatically create PRs when a dependency update (supports GitHub action, npm, and cargo) is available. I use it in my personal projects, and it works like a charm! And it's free for open-source projects!

File: .github/workflows/check-code.yml [hunk 1]
```diff
@@ -29,11 +30,12 @@ jobs:
         working-directory: ./web
```
Comment (pbrissaud) [inline .github/workflows/check-code.yml:32]: Why build the app after testing? I think it's useless

**Comment by pbrissaud** [comment]
I find checking both the PR title and all commits for adherence to conventional commits redundant, especially when using squash merges. Squash merging results in all changes combined into a single commit with the PR title, making individual commit checks unnecessary. Conversely, if using merge or rebase, the separate commit messages are visible, rendering the PR title check redundant. Simplifying the checks to align with the merge strategy would improve efficiency and avoid unnecessary duplication.

**Comment by ferdodo** [comment]
@pbrissaud 
image
Some messages are squashed and others kept unchanged

**Comment by pbrissaud** [comment]
> Some messages are squashed and others kept unchanged

Hmm, I didn't know that it's possible to have both at the same time 🤔 Would it not be better to define and enforce a single merging strategy?

File: .github/workflows/check-code.yml (hunk 1 above)
Comment (ferdodo) [reply .github/workflows/check-code.yml:32]: it was intended for running typechecking, I guess I should run the "check" script instead.

**Comment by ferdodo** [comment]
I just updated,

- PR title is now checked against the configuration file.
- I removed the PR description check.
- the 'check' script is run instead of the 'build' script.
- setup-node shall run the node version specified in package.json

**Review by pbrissaud — APPROVED** [review]
LGTM

**Pull request merged**